hack/generator/pkg/astmodel/storage_conversion_factories.go

```diff
@@ -36,23 +35,25 @@ var typeConversionFactories []StorageTypeConversionFactory
 
 func init() {
 	typeConversionFactories = []StorageTypeConversionFactory{
-		// Primitive types
+		// Primitive types and aliases
 		assignPrimitiveTypeFromPrimitiveType,
-		assignPrimitiveTypeFromOptionalPrimitiveType,
+		assignAliasedPrimitiveTypeFromAliasedPrimitiveType,
 		// Collection Types
 		assignArrayFromArray,
 		assignMapFromMap,
 		// Enumerations
 		assignEnumTypeFromEnumType,
 		assignPrimitiveTypeFromEnumType,
-		assignEnumTypeFromOptionalEnumType,
-		assignPrimitiveTypeFromOptionalEnumType,
 		// Complex object types
 		assignObjectTypeFromObjectType,
-		assignObjectTypeFromOptionalObjectType,
+		// Known types
+		assignKnownReferenceFromKnownReference,
 		// Meta-conversions
+		assignFromOptionalType, // Must go before assignToOptionalType
```

Comment:
might be worth noting why?

Reply:
If we don't to `FromOptionalType` first, we don't generate the correct zero value for the destination type.

For example, instead of this correct code:

``` go
	// age
	if source.age != nil {
		age := Age(*source.age)
		person.age = &age
	} else {
		person.age = nil
	}
```

we generate this incorrect result:

``` go
	// age
	if source.age != nil {
		ageTemp := Age(*source.age)
		person.age = &ageTemp
	} else {
		ageTemp := 0
		person.age = &ageTemp
	}
```

I've updated the comment to give a clue why the order is important.